subsystem/kubeapi_test.go

```diff
@@ -855,6 +870,84 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		}, "2m", "2s").Should(Equal(0))
 	})
 
+	FIt("deploy CD with ACI and agents with ignitionEndpoint - wait for ready, delete CD and verify ACI and agents deletion", func() {
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+		caCertificateSecretName := "ca-certificate"
+		deploySecret(ctx, kubeClient, caCertificateSecretName, map[string]string{corev1.TLSCertKey: "abc"})
+		modifiedAciSpec := aciSpec.DeepCopy()
+		modifiedAciSpec.IgnitionEndpoint = &hiveext.IgnitionEndpoint{
+			Url: "example.com",
+			CaCertificateReference: &hiveext.CaCertificateReference{
+				Namespace: Options.Namespace,
+				Name:      caCertificateSecretName,
+			},
+		}
+		deployAgentClusterInstallCRD(ctx, kubeClient, modifiedAciSpec, clusterDeploymentSpec.ClusterInstallRef.Name)
+		clusterKey := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+		infraEnvKey := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      infraNsName.Name,
+		}
+		infraEnv := getInfraEnvFromDBByKubeKey(ctx, db, infraEnvKey, waitForReconcileTimeout)
+		configureLocalAgentClient(infraEnv.ID.String())
+		hosts := make([]*models.Host, 0)
+		ips := hostutil.GenerateIPv4Addresses(3, defaultCIDRv4)
+		for i := 0; i < 3; i++ {
+			hostname := fmt.Sprintf("h%d", i)
+			host := registerNode(ctx, *infraEnv.ID, hostname, ips[i])
+			hosts = append(hosts, host)
+		}
+		for _, host := range hosts {
```

Comment:
No need for all the stuff done from here.
I think that what we want to test is:

- Create ACI with Ignition URL and cert
- Load cluster from DB and make sure that the URL and Cert were persisted
- Add Host
- Update Host spec with ignition token
- Load host from DB and make sure token was persisted.